### PR TITLE
Enable line wrapping

### LINKED ISSUE #706: Allow word wrap in `git diff` middle panel.
It's frustrating to view otherwise:

> 20240227T190724GMT

It would probably also be important for https://github.com/Murmele/Gittyup/issues/684#issue-2080855852.

### PR DESCRIPTION
**Pull request by jensenr30**
Line wrapping can be enabled/disabled with a setting in `Tools->Options->Diff->Wrap Lines`.  This closes #706.

All credit to Scintilla.

This addresses the second half of #514.  The second half of issue 514 was: bad user experience when scrolling down through the DiffView.  But with line wrapping enabled, the horizontal scroll bars disappear.  That prevents the mouse scroll wheel from being captured by the horizontal scroll bar.

Demo:

[wrap-lines.webm](https://github.com/Murmele/Gittyup/assets/3083835/f44666d1-76d9-4832-a78a-f89df74e0af4)

[wrap-lines-scroll.webm](https://github.com/Murmele/Gittyup/assets/3083835/967e653d-5d19-4df5-9167-24e6af63a71d)

### CODE REVIEW

**Comment by Murmele** [comment]
Thank you very much for the change!

File: src/conf/Settings.cpp [hunk 1]
```diff
@@ -191,6 +192,14 @@ QString Settings::hotkey(const QString &action) const {
   return value("hotkeys/" + action, "").toString();
 }
 
+bool Settings::isTextEditorWrapLines() const {
+  return value(kWrapLines).toBool();
+}
+
+void Settings::setTextEditorWrapLines(bool wrap) {
+  setValue(kWrapLines, wrap, true);
```
Comment (Murmele) [inline src/conf/Settings.cpp:200]: Hmm I am not sure if we should change the default behaviour.

File: src/editor/TextEditor.cpp [hunk 2]
```diff
@@ -76,6 +76,8 @@ TextEditor::TextEditor(QWidget *parent) : ScintillaIFace(parent) {
   setScrollWidthTracking(true);
   setSizeAdjustPolicy(QAbstractScrollArea::AdjustToContents);
 
+  setWrapMode(SC_WRAP_WORD);
+
```
Comment (Murmele) [inline src/editor/TextEditor.cpp:80]: Hmm I am not sure if we should change the default behaviour.

File: src/conf/Settings.cpp (hunk 1 above)
Comment (jensenr30) [reply src/conf/Settings.cpp:200]: I think `true` refers to the third argument which is "refresh".  I think it has to do with forcing QT to update stuff, but I'm not sure.

In any case, I tested with this code as-is, and the default behavior does not wrap lines.  So this should be OK.

File: src/editor/TextEditor.cpp (hunk 2 above)
Comment (jensenr30) [reply src/editor/TextEditor.cpp]: Ok.  I changed this line of code to `setWrapMode(SC_WRAP_NONE);`.  I tested this by running gittyup after `rm ~/.config/gittyup.github.com/Gittyup.conf ` and Line wrapping is disabled by default.

**Comment by jensenr30** [comment]
@Murmele 
> Can you add an entry to the changelog: https://github.com/Murmele/Gittyup/blob/master/docs/changelog.md You can add your gif also to the website when you add an entry at: https://github.com/Murmele/Gittyup/blob/master/docs/index.md.

Done!  I recorded a better video for the website:

[line-wrap-demo-2.webm](https://github.com/Murmele/Gittyup/assets/3083835/ce71236d-74b8-4f67-9e55-657ff2f184fc)

**Comment by Murmele** [comment]
Thanks I hope I can have a look into it next days.

File: docs/changelog.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ Description
 
 * UI(Commit List): Added a right-click menu entry to rename branches.
 * UI(Main Menu): Added a menu-entry to rename the current branch.
+* UI(Diff View): Added line wrapping option in Tools->Options->Diff
```
Comment (Murmele) [inline docs/changelog.md:9]:
```suggestion
* UI(Diff View): Added line wrapping option in Tools - Options - Diff
```
I think the `>` makes problems

**Review by Murmele — CHANGES_REQUESTED** [review]

**Review by Murmele — APPROVED** [review]

**Pull request merged**